docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+
+### WakuMessageHistoryArchive
+
+The `from` field SHOULD contain a timestamp of the time range's lower bound.
+
+The `to` field SHOULD contain a timestamp of the time range's the higher bound.
+
+The `contentTopic` field MUST contain the same `contentTopic` that the archive's `messages` have.
+
+The `messages` field MUST contain all messages that belong into the archive given its `from`, `to` and `contentTopic` fields.
+
+```
+syntax = "proto3"
+
+message WakuMessageArchive {
+  uint64 from = 1
+  uint64 to = 2
+  string contentTopic = 3
+  repeated WakuMessage messages = 4 // `WakuMessage` is provided by 14/WAKU2-MESSAGE
+}
+```
+
+## Creating magnet links
+
+Once a message archive is created, the community owner node MUST derive a magnet link following the [Magnet URI scheme](https://en.wikipedia.org/wiki/Magnet_URI_scheme) using the underlying BitTorrent protocol client.
+
+The `dn` parameter ("display name") in the resulting magnet link MAY be optional.
+
+The resulting magnet link MUST be bundled into a `WakuMessageArchiveIndex`, which is then later distributed to other Status nodes.
+
+## Bundling history archives into archive indices
+
+Community owner nodes MUST provide message archives for the entire community history. However, each individual archive only contains a subset of the complete history, that is, either data for a time range of seven days, or, a time range in which the node was offline. Therefore, message history archives need to be bundled into a `WakuMessageArchiveIndex`, which later distributed via the Waku network and allows receiving nodes to fetch archives for individual time ranges.
+
+A `WakuMessageArchiveIndex` is a map where the key is the KECCAK-256 hash of the magnet link derived from the previously created archive and the value an instance of `WakuMessageArchiveMetadata`, which is also derived from the previously created message archive.
+
+### WakuMessageArchiveMetadata
+
+The `from` field SHOULD contain a timestamp of the time range's lower bound.
+
+The `to` field SHOULD contain a timestamp of the time range's the higher bound.
+
+The `contentTopic` field MUST contain the same `contentTopic` that the archive's `messages` have.
+
+The `magnet_uri_hash` field MUST contain the KECCAK-256 hash of the magnet link created from the `WakuMessageArchive`.
+
+The `magnet_uri` field MUST contain the magnet link created from the `WakuMessageArchive`.
+
+```
+syntax = "proto3"
+
+message WakuMessageArchiveMetadata {
+  uint64 from = 1
+  uint64 to = 2
+  string contentTopic = 3
+  string magnet_uri_hash = 4
+  string magnet_uri = 5
+}
+
+message WakuMessageArchiveIndex {
+  map<string, WakuMessageArchiveMetadata> archives = 1
+}
+```
+
+The community owner node MUST create a `WakuMessageArchiveIndex` every time it creates a new `WakuMessageArchive`.
+
+For every created `WakuMessageArchive`, there MUST be a `WakuMessageArchiveMetadata` entry in the index map.
```

Comment:
I don't understand why all of this is needed, and why we don't just sent the latest magnet link over the hidden channel whenever it's created.  The latest magnet link will always be the single thing the user needs to download the entire history of the communtiy